lib/Parser.js

```diff
@@ -21,28 +21,19 @@ const joinRanges = (startRange, endRange) => {
 	return [startRange[0], endRange[1]];
 };
 
-const ECMA_VERSION = 2017;
-
-const POSSIBLE_AST_OPTIONS = [{
+const defaultParserOptions = {
 	ranges: true,
 	locations: true,
-	ecmaVersion: ECMA_VERSION,
+	ecmaVersion: 2017,
 	sourceType: "module",
+	onComment: null,
 	plugins: {
 		dynamicImport: true
 	}
-}, {
-	ranges: true,
-	locations: true,
-	ecmaVersion: ECMA_VERSION,
-	sourceType: "script",
-	plugins: {
-		dynamicImport: true
-	}
-}];
+};
 
 class Parser extends Tapable {
-	constructor(options) {
+	constructor(options, type = "javascript/auto") {
```

Comment:
I think the parser shouldn't know about different module types. Pass sourceType directly.